usr/share/rear/lib/rsync-functions.sh

```diff
@@ -0,0 +1,163 @@
+# Functions for manipulation of rsync URLs (both OUTPUT_URL and BACKUP_URL)
+
+#### OLD STYLE:
+# BACKUP_URL=[USER@]HOST:PATH           # using ssh (no rsh)
+#
+# with rsync protocol PATH is a MODULE name defined in remote /etc/rsyncd.conf file
+# BACKUP_URL=[USER@]HOST::PATH          # using rsync
+# BACKUP_URL=rsync://[USER@]HOST[:PORT]/PATH    # using rsync (is not compatible with new style!!!)
+
+#### NEW STYLE:
+# BACKUP_URL=rsync://[USER@]HOST[:PORT]/PATH    # using ssh
+# BACKUP_URL=rsync://[USER@]HOST[:PORT]::/PATH  # using rsync
+
+function rsync_validate () {
+    local url="$1"
+
+    if [[ "$(url_scheme "$url")" != "rsync" ]]; then # url_scheme still recognizes old style
```

Comment:
Could you also update the comment in the
url_scheme() function to
```
    # rsync scheme does not have to start with rsync:// it can also be scp style
    # see the comments in usr/share/rear/lib/rsync-functions.sh
```

Reply:
Good catch. This made me test the "scp-style", and also the alternative old-style rsync url (with double `::` and no `rsync://` prefix). Both work.

Reply:
Done in 2008a537ec5c152438488b849c72de4b2e36f141